crates/vendor/morphic_lib/src/analyze.rs

```diff
@@ -357,8 +359,8 @@ impl<'a> ForwardState<'a> {
     }
 
     fn touch(&mut self, version: BackRefStateVersionId, heap_cell: HeapCellId) {
-        let back_refs = std::mem::take(self.back_refs(version, heap_cell));
-        for &query_point in &back_refs {
+        let mut back_refs = std::mem::take(self.back_refs(version, heap_cell));
+        for query_point in back_refs.drain() {
```

Comment:
This is the one change I am not completely sure of. I added a drain here to clear out the QueryPoints when setting the fate to DirectTouch.

Reply:
but... `mem::take` already moves the `QueryPoints` out of that mutable reference that `back_refs` returns. The value is replaced with an empty hash set. 

So this drain is pointless, because you drain the whole iterator anyway. `into_iter` would do the same thing, and just `iter` does the same thing too in practice because the query points are copy.

am I missing something?

Reply:
Oh, that's fair, I guess I just need to delete the line after the for loop that reinserts the back_refs into the main map:
```
*self.back_refs(version, heap_cell) = back_refs;
```

I'll update to that instead.

Reply:
Oh, though I think the current code would reuse the allocatoin, so it may be better?

Reply:
Ok. actually the change is about 30s faster. Down to 2.5min. Apparently `into_iter` is much faster than `drain` here.